DACT-558 - altering code so that we do not try and build termination …

rwalrondch opened the pull request:
…filing data when we are doing an AP01,  as that breaks as there is no officer appointment as it is a new officer.

AlisonGriffiths commented on src/main/java/uk/gov/companieshouse/officerfiling/api/service/FilingDataServiceImpl.java:
```diff
@@ -70,22 +70,25 @@ public FilingApi generateOfficerFiling(String transactionId, String filingId, St
 
-    private void setFilingApiData(FilingApi filing, String transactionId, String filingId,
-                                  String ericPassThroughHeader) {
+    private void setTerminationFilingApiData(FilingApi filing, String transactionId, String filingId,
```
Do you need the FilingApi filing method param here? Doesn't seem to be used.

rwalrondch replied:
I am assuming you mean on the setAppointmnetFilingApiData rather than here on Termination,  yes it's needed for Termination, but I have removd it for Appointment.

AlisonGriffiths commented on src/main/java/uk/gov/companieshouse/officerfiling/api/service/FilingDataServiceImpl.java:
```diff
@@ -119,6 +122,18 @@ private void setFilingApiData(FilingApi filing, String transactionId, String fil
+            String ericPassThroughHeader) {
+        //TODO - just creating a blank one of these for the moment,  need to add filing data into here as we add it in.
+        final var transaction = transactionService.getTransaction(transactionId, ericPassThroughHeader);
+        // String companyNumber = transaction.getCompanyNumber();
```
you could delete the commented out code here?

AlisonGriffiths reviewed (changes requested):
The sonarqube report hasn't run but there are a couple of sonar issues on this file.
Also, do you need to update the associated test class for this file?

rwalrondch commented:
> The Sonarqube report hasn't run but there are a couple of sonar issues on this file. Also, do you need to update the associated test class for this file?

I have added a test class for the new AP01 flow,  it doesn't check for much though as its not doing anything yet, but it does check it goes down the AP01 route. the test will be flushed out when we get to doing the Filing data Story.

also i fixed a couple of sonar moans, but one seemed to be about a field isn't used when it actually is(filingDescription), I think Sonar can't tell correctly in the way that it is used, maybe a spring thing?
